### ISSUE TITLE
v2.0.0 Release :ship: Shipping the Revamp :ship:

### ISSUE THREAD

**Issue by olizilla** [ipfs-shipyard/ipfs-webui#749]
### What needs to happen to get the revamp branch released

- [ ] Status page with node identity info and node bandwidth chart.
- [ ] Integrate file share app. https://github.com/ipfs-shipyard/ipfs-share-files
- [ ] Export and reuse [ipld-explorer](https://github.com/ipfs-shipyard/ipld-explorer)
- [ ] Polish files page. https://github.com/ipfs-shipyard/ipfs-webui/issues/748
- [ ] Improve perf of peer bandwidth stats (or disable it until a better api is available)
- [ ] Finish API address config page PR https://github.com/ipfs-shipyard/ipfs-webui/pull/715
- [ ] https://github.com/ipfs-shipyard/ipfs-webui#release-a-new-version-of-the-webui
- [ ] Disable the CRA service worker, until we have the update mechanism in place.
- [ ] Move the revamp branch to master, bump package.json version to 2.0.0, and tag it.
- [ ] Pubish to ipfs and release it. https://github.com/ipfs-shipyard/ipfs-webui#release-a-new-version-of-the-webui

### Work that will wait until v2.1

Start tackling these after the revamp becomes the new master branch and is tagged for a release.

- [ ] Define a better deployment and update mechanism. IPLD Explorer announces when a new version is available. We could publish Web UI under a DNSLINK address, and have the app check a release version to see if there is a newer version (https://github.com/ipfs-shipyard/ipld-explorer/pull/21#issuecomment-409902600)
- [ ] File availability indicator. https://github.com/ipfs-shipyard/ipfs-webui/pull/718 asking for providers every file from every webui user will be slow, resource intensive and could spam the network, so this needs some more thought.

**Comment by hacdias** [comment]
@olizilla I think we should add i18n to 2.1 😄

**Comment by lidel** [comment]
Hm.. why wait for 2.1? We don't need to have unified crowdsourcing website set up for 2.0 (https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/50), but we should ship i18 files with English locale from the start.
That way if someone wants to contribute, there would be an option to create a PR on github.

**Comment by olizilla** [comment]
I agree with @lidel, let's at least get the i18n plumbing in place, and add the en-us file so we can point people at it if they ask. I'll add am item to the list in this issue.

So yes, there is a chunk of code changes needed to go an replace all the hardcoded strings with intl look ups, but we need to do that anyway, and things are settling down, so let's do it! To be considerate to the work already done for the current version of webui, we should look at the language files that exist already, and see what entries can be re-used in the revamp as is. 

**Bonus points** for reviewing our use of language for consistency, and making sure that we're sticking to the same words for the same concept everywhere. 

**Double bonus points** for identifying where we already have a good phrase for things in the existing webui and re-using in the new one if it still makes sense, though that is a trade-off, as we do want to take the opportunity to improve the wording, but if there are phrases that are just as good in the old version, then let's reuse the one we have translations for already.

**Triple bonus points** for cross referencing all the language with ipfs-companion! We probably have some already translated phrases in there that can be reused.

Leave the explore section out initially. I need to talk through the options for re-using it with you all tomorrow, and we can decide how i18n will work there after that.

**Comment by hacdias** [comment]
Sure, then we need to decide which library to use before starting working on it.

**Comment by olizilla** [comment]
@hacdias didn't you already use react-intl?

**Comment by hacdias** [comment]
@olizilla actually we did, yes. Didn't remember. I'll check how it works with external files.

**Comment by olizilla** [comment]
I'm gonna update the list with
- Upgrade all UI text to use ~~react-intl~~ -> i18next
- handle ipfs connection errors.

**Comment by olizilla** [comment]
We are super close now. We're waiting on the 

- `webui.ipfs.io` subdomain to be provisioned
- #802 to land to inform users when IPFS goes away.

We're going to bump
- [ ] Integrate file share app. https://github.com/ipfs-shipyard/ipfs-share-files/issues/11
due to the file upload blocker in https://github.com/ipfs/js-ipfs-api/issues/797 

...of note this will impact web-ui as well, but adding multiple files is the core feature of share, where as the webui files view is useful without support for large / multi file upload.

**Comment by olizilla** [comment]
the `master` branch is now `revamp`. Cool original flavour webui lives on at the `v1` branch https://github.com/ipfs-shipyard/ipfs-webui/tree/v1

Next stop. Continuous Deployment of master to https://webui.ipfs.io in #806
